src/uri.cr

```diff
@@ -221,7 +221,11 @@ class URI
   # uri.request_target # => "/?foo=bar"
   # ```
   def request_target : String
-    String.build do |str|
+    string_size = 1
+    string_size += @query ? @query.not_nil!.bytesize : -1
```

Comment:
I think better way to deal with nillable properties is something like
```if q = @query 
  string_size += q.bytesize 
else
  string_size -= 1
end
```
On the other hand, procedure isn't thread-safe anyway.